### DISCUSSION TITLE
Adding an external .gds file as a cell

### DISCUSSION THREAD

**Discussion by Diego-planckian1** [iqm-finland/KQCircuits#87] [Q&A] [answered]
Hi,

my company is working with an external research center ( VTT indeed :) ), which provides us a .gds file that has to be inserted as a cell without modifications (geometries, layers, ect.). How can I do it within the KQCircuits package?

**Comment by qpavsmi** [comment]
If the external .gds file is an entire chip that you want to include in the wafer, then it is as simple as giving a filepath string in the mask script.

But seems that you want to include an element within the KQCircuits chip? I've given instructions in the past about using a .gds file to insert a qubit in the chip here: https://github.com/iqm-finland/KQCircuits/discussions/83#discussioncomment-8380850 Let me know if this works for you. In summary, place the .gds file either in `klayout_package/python/kqcircuits/qubits` if it is specifically a qubit or to `klayout_package/python/kqcircuits/elements` if it is just an element. Then using the code snippet you should be able to make KQC fetch that geometry and place it at the needed location.

**Reply by Diego-planckian1** [reply]
Yes it is just an element. Excuse me if I haven't find it in previous discussions.

**Reply by qpavsmi** [reply]
No worries :) Hopefully this approach helps you too. You said you want to use exact same layer definitions as in the .gds file, but if you want to move them to KQC layers or it somehow interferes with KQC layer definitions then let me know, the polygons can be easily moved from layer to layer,

**Reply by Diego-planckian1** [reply]
It doesn't seem to work. The discussion you link was for an .oas file... maybe that is the problem?

**Reply by qpavsmi** [reply]
I didn't have issues if I mixed .oas and .gds files before...

Are you able to see your element listed on the Element Library panel (bottom left in KLayout) and are you able to drag-and-drop it into the layout?

**Reply by Diego-planckian1** [reply]
I saved the wanted file as .oas and it worked!

the sizes are wrong... what can i modify?

**Reply by qpavsmi** [reply, marked as answer]
OK, good to know that the file format was the issue :D

After `self.layout.create_cell` and before `self.insert_cell`, try adding this line:
```
qubit_cell = self.layout.create_cell("somequbit", Qubit.LIBRARY_NAME)
qubit_cell.transform(pya.DCplxTrans(0.1))  # shrinks to 10% (0.1) of the size
self.insert_cell(qubit_cell, pya.DTrans(0, False, 5000, 5000), "qb_0")
```

**Reply by Diego-planckian1** [reply]
It works! thank you again!
